functions/getEmployees/index.ts

```diff
@@ -0,0 +1,52 @@
+import { AzureFunction, Context, HttpRequest } from "@azure/functions";
+import fetch from "node-fetch";
+
+const httpTrigger: AzureFunction = async function (
+  context: Context,
+  req: HttpRequest
+): Promise<void> {
+  if (
+    process.env.NODE_ENV !== "production" &&
+    !process.env.CV_PARTNER_API_SECRET
+  ) {
+    context.res = {
+      status: 500,
+      body: `Environment variable CV_PARTNER_API_SECRET is missing.
+Go to cv-partner and get an API key, add under "Values" in local.settings.json`,
+    };
+    return;
+  }
+  try {
+    const request = await fetch("https://variant.cvpartner.com/api/v1/users", {
+      headers: [
+        ["Authorization", `Bearer ${process.env.CV_PARTNER_API_SECRET || ""}`],
+      ],
+    });
+    if (request.ok) {
+      const employeesJSON = await request.json();
+      // Make images
+      const employeeList = employeesJSON.filter(
+        (employee) =>
+          employee.deactivated ||
+          // All employees that have started should be set
+          // as countrymanager in cv-parter, when they start.
+          employee.roles.some((role) => role === "countrymanager")
+      );
+      context.res = {
+        body: employeeList,

```

Comment:
Should we limit the amount of data we pass on to just what we need? In this case just the images. Both to limit data so we don't expose things we don't intend to. But also to reduce payloads.

Reply:
Had the same thought, but came to the conclusion that it would be easier to just add an extra layer of data massage. Excess data are handled during the `.getStaticProps` so user should never be able to access that data we don't want to exopse (unless we start to use the api client-side). Are we paying for bandwith on Vecrel?, on Azure it's runtime so its cheaper to not do extra operations on the data.

Reply:
Not paying for bandwidth. I think the minor transformation cost on CPU is neglectable for functions also, tbh. This is fairly theoretical and somewhat loaded of paranoia, but given that you expose all data, you could technically push a PR passing on the entire data set on a branch and log that. I don't think it is likely though, but yeah.

Reply:
I understand what you mean, and maybe was part of the reason for moving it behind a proxy in the first place.